packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Link.tsx

```diff
@@ -0,0 +1,236 @@
+import * as React from 'react';
+
+import {
+  BaseLink,
+  Button,
+  Field,
+  FieldInput,
+  FieldLabel,
+  Flex,
+  Icon,
+  Popover,
+  Tooltip,
+  Typography,
+} from '@strapi/design-system';
+import { Pencil, Trash } from '@strapi/icons';
+import { useIntl } from 'react-intl';
+import { Editor, Path, Range, Transforms } from 'slate';
+import { type RenderElementProps, ReactEditor } from 'slate-react';
+import styled from 'styled-components';
+
+// @ts-expect-error TODO migrate this file
+import { composeRefs } from '../../../utils';
+import { type BlocksStore, useBlocksEditorContext } from '../BlocksEditor';
+import { editLink, removeLink } from '../utils/links';
+import { type Block } from '../utils/types';
+
+const StyledBaseLink = styled(BaseLink)`
+  text-decoration: none;
+`;
+
+// Make sure the tooltip is above the popover
+const TooltipCustom = styled(Tooltip)`
+  z-index: 6;
+`;
+
+// Used for the Edit and Cancel buttons in the link popover
+const CustomButton = styled(Button)`
+  & > span {
+    line-height: normal;
+  }
+`;
+
+const Link = React.forwardRef<HTMLAnchorElement, RenderElementProps>(
+  ({ element, children, attributes }, forwardedRef) => {
+    const { formatMessage } = useIntl();
+    const { editor } = useBlocksEditorContext('Link');
+    const path = ReactEditor.findPath(editor, element);
+    const [popoverOpen, setPopoverOpen] = React.useState(
+      editor.lastInsertedLinkPath ? Path.equals(path, editor.lastInsertedLinkPath) : false
+    );
+
+    const elementAsLink = element as Block<'link'>;
```

Comment:
I'm not sure if you wrote this or not, but i think a typeguard might be more appropriate?